cpp/log/cert_submission_handler.cc

```diff
@@ -18,6 +18,30 @@ using ct::LogEntry;
 using ct::PrecertChainEntry;
 using ct::X509ChainEntry;
 using std::string;
+using util::Status;
+
+namespace {
+
+Status GetVerifyError(CertChecker::CertVerifyResult result) {
+  switch (result) {
+    case CertChecker::INVALID_CERTIFICATE_CHAIN:
+    case CertChecker::PRECERT_EXTENSION_IN_CERT_CHAIN:
+    case CertChecker::UNSUPPORTED_ALGORITHM_IN_CERT_CHAIN:
```

Comment:
I wondered if we needed to break out UNSUPPORTED_ALGORITHM separately, possibly add it to util::Status? A couple of places check for this explicitly, though it's possible they then return a generic error anyway.